src/MercuryMock.ts

```diff
@@ -0,0 +1,208 @@
+import MercuryAdapterMock from './adapters/MercuryAdapterMock'
+import log from './lib/log'
+import Mercury, { MercuryAdapterKind } from './Mercury'
+import {
+	IMercuryEventContract,
+	IOnData,
+	ISkillOnData,
+	IMercuryEmitOptions,
+	OnHandler
+} from './types/mercuryEvents'
+import { MercurySubscriptionScope } from './types/subscriptions'
+
+let faker: {
+	name: {
+		title: () => string
+	}
+}
+try {
+	faker = require('@sprucelabs/test').faker
+} catch (e) {
+	faker = {
+		name: {
+			title: () => 'testing title'
+		}
+	}
+}
+
+interface IMockEmitResponses {
+	[eventName: string]: {
+		[skillId: string]: Omit<IOnData<any, any, any>, 'eventId'>
+	}
+}
+
+export default class MercuryMock<
```

Comment:
is this needed or can tests depend solely on MockAdapter.

the reason I ask is `emit` is being overwritten from the base class, which essentially means the base class is not truly being tested.

if the mock functionality could be consolidated to the adapter (I'd have to better understand what is being done before saying it could absolutely be done), we would, as a result, have better tests. It would also be more cohesive if the mock functionality was consolidated to the mock adapter.